docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Recommendation/OneClassMatrixFactorizationWithOptions.cs

```diff
@@ -0,0 +1,144 @@
+        // one-class matrix factorization.
+        // In this example we will create in-memory data and then use it to train a one-class matrix factorization model.
+        // Afterward, prediction values are reported.
+        // To run this example, it requires installation of additional nuget package
+        // <a href="https://www.nuget.org/packages/Microsoft.ML.Recommender/">Microsoft.ML.Recommender</a>.
+        public static void Example()
+        {
+            // Create a new context for ML.NET operations. It can be used for exception tracking and logging,
+            // as a catalog of available operations and as the source of randomness.
+            var mlContext = new MLContext(seed: 0);
+
+            // Get a small in-memory dataset.
+            GetOneClassMatrix(out List<MatrixElement> data, out List<MatrixElement> testData);
+
+            // Convert the in-memory matrix into an IDataView so that ML.NET components can consume it.
+            var dataView = mlContext.Data.LoadFromEnumerable(data);
+
+            // Create a matrix factorization trainer which may consume "Value" as the training label, "MatrixColumnIndex" as the
+            // matrix's column index, and "MatrixRowIndex" as the matrix's row index. Here nameof(...) is used to extract field
+            // names' in MatrixElement class.
+            var options = new MatrixFactorizationTrainer.Options
+            {
+                MatrixColumnIndexColumnName = nameof(MatrixElement.MatrixColumnIndex),
+                MatrixRowIndexColumnName = nameof(MatrixElement.MatrixRowIndex),
+                LabelColumnName = nameof(MatrixElement.Value),
+                NumberOfIterations = 20,
+                NumberOfThreads = 8,
+                ApproximationRank = 32,
+                Alpha = 1,
+                // The desired of unobserved values.
+                C = 0.15,
+                // To enable one-class matrix factorization, the following line is required.
+                LossFunction = MatrixFactorizationTrainer.LossFunctionType.SquareLossOneClass
+            };
+
+            var pipeline = mlContext.Recommendation().Trainers.MatrixFactorization(options);
+
+            // Train a matrix factorization model.
+            var model = pipeline.Fit(dataView);
+
+            // Apply the trained model to the test set. Notice that training is a partial 
+            var prediction = model.Transform(mlContext.Data.LoadFromEnumerable(testData));
+
+            var results = mlContext.Data.CreateEnumerable<MatrixElement>(prediction, false).ToList();
+            // Feed the test data into the model and then iterate through a few predictions.
+            foreach (var pred in results.Take(15))
+                Console.WriteLine($"Predicted value at row {pred.MatrixRowIndex - 1} and column {pred.MatrixColumnIndex - 1} is {pred.Score} and its expected value is {pred.Value}.");
```

Comment:
>Console.Write [](start = 16, length = 13)

Break this line; I would suggest for such long lines using old-style `{0}...{1}` string formatting so you can throw the arguments onto new lines.

Reply:
This is long but makes the code readable like a text message.

---
In reply to: [274576483](https://github.com/dotnet/machinelearning/pull/3282#discussion_r274576483) [](ancestors = 274576483)